### ISSUE TITLE
`InvalidRestrictedOrder` error when fulfilling an order created via sdk.`createSellOrder`  on testnet

### ISSUE THREAD

**Issue by mr-friend** [ProjectOpenSea/opensea-js#744]
The order for this NFT was created using sdk.createSellOrder(...):
[https://testnets.opensea.io/assets/goerli/0xa7e3322f6d77c7c768c313dd75438f8954a4ad40/2](https://testnets.opensea.io/assets/goerli/0xa7e3322f6d77c7c768c313dd75438f8954a4ad40/2)

It cannot be fulfilled neither via the UI, nor via the sdk.fulfillOrder(...)

sdk init:
>var provider = new HDWalletProvider({
    privateKeys: [keys.OWNER_WALLET_PRIVATE_KEY], // wallet private key
    providerOrUrl: keys.WEB3_RPC_URL, //goerli infura URL here
    chainId: keys.WEB3_NETWORK_CHAINID // 5
});
const web3 = new Web3(provider);
web3.eth.accounts.wallet.add(keys.OWNER_WALLET_PRIVATE_KEY);
web3.eth.defaultAccount = web3.eth.accounts.wallet[0].address;

>const openseaSDK = new OpenSeaSDK(provider, { networkName: 'goerli', apiKey: null, useReadOnlyProvider: false }, console.debug);

createSellOrder:
>var expirationTime = Math.round(Date.now() / 1000 + 60 * 60 * 24 * 90);
var sellPrice = '0.0001';
var listing = await openseaSDK.createSellOrder({
    asset: {
        tokenId: '2',
        tokenAddress: '0xa7e3322f6d77c7c768c313dd75438f8954a4ad40',
    },
    accountAddress: '0x60fBDE0dED533Aa37c6b26DF81c318DfB8702686',
    startAmount: sellPrice,
    expirationTime: expirationTime,
})

ERROR when trying to fulfill via UI (MetaMask):
>cannot estimate gas; transaction may fail or may require manual gas limit [ See: https://links.ethers.org/v5-errors-UNPREDICTABLE_GAS_LIMIT ] (reason="execution reverted", method="estimateGas", transaction={"from":"0x197D37d1e6f321209d01b4Fa837c228902e1D7d2","to":"0x00000000006c3852cbEf3e08E8dF289169EdE581","value":{"type":"BigNumber","hex":"0xbefe6f672000"},"data":"0xfb0f3ee10000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000ba3813048c0000000000000000000000000060fbde0ded533aa37c6b26df81c318dfb87026860000000000000000000000000000000000000000000000000000000000000000000000000000000000000000a7e3322f6d77c7c768c313dd75438f8954a4ad40000000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000003000000000000000000000000000000000000000000000000000000006323882700000000000000000000000000000000000000000000000000000000639a2f213000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001a8cc26f2bed643312dabd0c066864030000007b02230091a7ed01230072f7006a004d60a8d4e71d599b8104250f00000000007b02230091a7ed01230072f7006a004d60a8d4e71d599b8104250f00000000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000024000000000000000000000000000000000000000000000000000000000000002a00000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000004c65c6294000000000000000000000000000000a26b00c1f0df003000390027140000faa7190000000000000000000000000000000000000000000000000000000000000040f41b5fde00611b48dde38a49d3afc2e7266fd02e8ae231fb64c4e22f1529f6061742e65d00126809f6db6bf8be3ae059afd225bf93834204a03148db238efbe8","accessList":null}, error={"code":-32603,"message":"execution reverted","data":{"originalError":{"code":3,"data":"0xfb5014fc5ad35f68b22dc9eef8af9afc2a3ad7bd6a56e0b49d3d79d20f10e35b2dc0c8b0","message":"execution reverted"}}}, code=UNPREDICTABLE_GAS_LIMIT, version=providers/5.7.1)

Decoded Input Data:
>  parameters: [
    considerationToken: '0x0000000000000000000000000000000000000000',
    considerationIdentifier: '0',
    considerationAmount: '204750000000000',
    offerer: '0x60fBDE0dED533Aa37c6b26DF81c318DfB8702686',
    zone: '0x0000000000000000000000000000000000000000',
    offerToken: '0xA7E3322f6D77c7C768C313Dd75438f8954a4AD40',
    offerIdentifier: '2',
    offerAmount: '1',
    basicOrderType: '3',
    startTime: '1663272999',
    endTime: '1671048993',
    zoneHash: '0x3000000000000000000000000000000000000000000000000000000000000000',
    salt: '35290793045986680091254317593737585667',
    offererConduitKey: '0x0000007b02230091a7ed01230072f7006a004d60a8d4e71d599b8104250f0000',
    fulfillerConduitKey: '0x0000007b02230091a7ed01230072f7006a004d60a8d4e71d599b8104250f0000',
    totalOriginalAdditionalRecipients: '1',
    additionalRecipients: [ [Array] ],
    signature: '0xf41b5fde00611b48dde38a49d3afc2e7266fd02e8ae231fb64c4e22f1529f6061742e65d00126809f6db6bf8be3ae059afd225bf93834204a03148db238efbe8'
  ]
}

Decoded Error Data:
>{
  name: 'InvalidRestrictedOrder',
  params: [
    {
      name: 'orderHash',
      value: '0x5ad35f68b22dc9eef8af9afc2a3ad7bd6a56e0b49d3d79d20f10e35b2dc0c8b0',
      type: 'bytes32'
    }
  ]
}

Returned orderHash is for a [different NFT all-together](https://testnets.opensea.io/assets/goerli/0x0e33f1b08cd86aca98e0d57b9ddd55d4683200c8/505):
>var order = await openseaSDK.api.getOrder('0x5ad35f68b22dc9eef8af9afc2a3ad7bd6a56e0b49d3d79d20f10e35b2dc0c8b0')
console.log(order.takerAssetBundle.assets)

>[
  {
    tokenId: '505',
    tokenAddress: '0x0e33f1b08cd86aca98e0d57b9ddd55d4683200c8',
    name: 'Mint Pass Mexico City | MPMX',
    description: 'MPMX',
    owner: {
      user: null,
      profile_img_url: 'https://storage.googleapis.com/opensea-static/opensea-profile/1.png',
      address: '0x0000000000000000000000000000000000000000',
      config: ''
    },
    assetContract: {
      name: 'Mint Pass Mexico City | MPMX',
      description: null,
      type: 'non-fungible',
      schemaName: 'ERC721',
      address: '0x0e33f1b08cd86aca98e0d57b9ddd55d4683200c8',
      tokenSymbol: 'MPMX',
      buyerFeeBasisPoints: 0,
      sellerFeeBasisPoints: 250,
      openseaBuyerFeeBasisPoints: 0,
      openseaSellerFeeBasisPoints: 250,
      devBuyerFeeBasisPoints: 0,
      devSellerFeeBasisPoints: 0,
      imageUrl: null,
      externalLink: null,
      wikiLink: undefined
    },
    collection: {
      createdDate: 2022-09-15T20:26:56.124Z,
      name: 'Mint Pass Mexico City  MPMX V2',
      description: null,
      slug: 'mint-pass-mexico-city-mpmx-v2-1',
      editors: undefined,
      hidden: false,
      featured: false,
      featuredImageUrl: null,
      displayData: [Object],
      paymentTokens: [],
      openseaBuyerFeeBasisPoints: 0,
      openseaSellerFeeBasisPoints: 250,
      devBuyerFeeBasisPoints: 0,
      devSellerFeeBasisPoints: 0,
      payoutAddress: null,
      imageUrl: null,
      largeImageUrl: null,
      stats: undefined,
      traitStats: undefined,
      externalLink: null,
      wikiLink: null,
      fees: [Object]
    },
    orders: null,
    sellOrders: null,
    buyOrders: null,
    isPresale: undefined,
    imageUrl: 'https://lh3.googleusercontent.com/y6rYAS_CBTgXb9UpwXYCNTxnlS_T6j7xFRGohF1urOGqKvIvSIt--Ut32EnIlYejOC7-KmIo3fGYfOz0M0_xkVaYmCy8hyyZkQSDTQ=s250',
    imagePreviewUrl: 'https://lh3.googleusercontent.com/y6rYAS_CBTgXb9UpwXYCNTxnlS_T6j7xFRGohF1urOGqKvIvSIt--Ut32EnIlYejOC7-KmIo3fGYfOz0M0_xkVaYmCy8hyyZkQSDTQ=s250',
    imageUrlOriginal: 'https://bh.mypinata.cloud/ipfs/QmT7jWJYm2STkrwrRLc7EKsdC5McqHhUAmzig6tKgbGmky',
    imageUrlThumbnail: 'https://lh3.googleusercontent.com/y6rYAS_CBTgXb9UpwXYCNTxnlS_T6j7xFRGohF1urOGqKvIvSIt--Ut32EnIlYejOC7-KmIo3fGYfOz0M0_xkVaYmCy8hyyZkQSDTQ=s128',
    externalLink: null,
    openseaLink: 'https://testnets.opensea.io/assets/goerli/0x0e33f1b08cd86aca98e0d57b9ddd55d4683200c8/505',
    traits: undefined,
    numSales: 0,
    lastSale: null,
    backgroundColor: null,
    transferFee: null,
    transferFeePaymentToken: null
  }
]

**Comment by taerbao** [comment]
same issue

**Comment by megan7654** [comment]
hello @mr-friend have you been able to fix this error? I will suggest you talk with an Opensea Developer concerning this issue,on their live chat Developers support page. (CLICK HERE) [OPENSEA SUPPORT PAGE](https://opensea.opns.support/) Using your TICKET ID: #619589
to engage in a conversation with opensea support and have the transaction looked into so as to resolve it right away.

**Comment by Irantwomiles** [comment]
I managed to fix this issue by doing exactly what it says, which is setting the gas limit manually. There is no real easy way to go about it other than reading the source code and testing out different changes in the built files of the @opensea node module.

Here's how I did it: 

Navigate to the @opensea module in your node_modules directory and then open the seaport-js/lib/utils/usecase.js file. In that file you find where `__generator` is being defined and file the `sent` function within `__generator`. From what I can tell, the `sent` function handles all of the gas and gas limit values. To modify the gas limit just set `t[1].gasLimit = yourGasLimitHere`. I hope that helps you out.
